projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -0,0 +1,875 @@
+                    .subscribe((date: Date) => {
+                        this.value = { start: this.value?.start, end: date as Date };
+                    });
+                start.dateTimeEditor.validationFailed
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe((event: IgxDateTimeEditorEventArgs) => {
+                        this.value = { start: event.newValue as Date, end: this.value?.end };
+                    });
+                end.dateTimeEditor.validationFailed
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe((event: IgxDateTimeEditorEventArgs) => {
+                        this.value = { start: this.value?.start, end: event.newValue as Date };
+                    });
+            }
+        }
+    }
+
+    private attachOnTouched(): void {
+        if (this.hasProjectedInputs) {
+            this.projectedInputs.map(i => {
+                fromEvent((i as IgxDateRangeInputsBaseComponent).dateTimeEditor.nativeElement, 'blur')
+                    .pipe(takeUntil(this.$destroy))
+                    .subscribe(() => this.onTouchCallback());
+            });
+        } else {
+            fromEvent(this.inputDirective.nativeElement, 'blur')
+                .pipe(takeUntil(this.$destroy))
+                .subscribe(() => this.onTouchCallback());
+        }
+    }
+
+    private configPositionStrategy(): void {
+        this._positionSettings = {
+            openAnimation: fadeIn,
+            closeAnimation: fadeOut,
+            target: this.element.nativeElement
+        };
+        this._dropDownOverlaySettings.positionStrategy = new AutoPositionStrategy(this._positionSettings);
+    }
+
+    private configOverlaySettings(): void {
+        if (this.overlaySettings !== null) {
+            this._dropDownOverlaySettings = Object.assign({}, this._dropDownOverlaySettings, this.overlaySettings);
+            this._dialogOverlaySettings = Object.assign({}, this._dialogOverlaySettings, this.overlaySettings);
+        }
+    }
+
+    private updateInputs(): void {
+        const start = this.projectedInputs?.find(i => i instanceof IgxDateRangeStartComponent) as IgxDateRangeStartComponent;
+        const end = this.projectedInputs?.find(i => i instanceof IgxDateRangeEndComponent) as IgxDateRangeEndComponent;
+        if (start && end && this.value) {
+            start.updateInput(this.value.start);
+            end.updateInput(this.value.end);
+        }
+    }
+
+    private valueInRange(value: DateRange): boolean {
+        this.normalizeDates();
+        if (this.minValue && this.maxValue) {
+            if (!value || !value?.start || !value?.end) { return false; }
+            return this.minValue.getTime() <= value.start.getTime() &&
+                value.end.getTime() <= this.maxValue.getTime();
+        }
+
+        return value?.start && this.minValue && this.minValue.getTime() <= value.start.getTime() ||
```

Comment:
what if we have no value?
```
undefined > date.getTime()
```
This will not work.
First if start is less than min -> return false.
Second if end is more than max -> return false.
Finally, we may have only start (IIRC we cannot have only end), if start > max -> return false.
In all other cases -> return true.
If there is no value I would throw - I am not sure if this is valid and we should not call `valueInRange` and pass `null`.
I would go with something like this:
```
if value is not null
   if we have minValue and start and start < minValue -> return false
   if we have maxValue and end and end > maxValue -> return false
   if we have maxValue and start and start > maxValue -> return false
  return true
```